tasks/main.yml

```diff
@@ -67,15 +67,53 @@
     - haproxy-configuration
     - haproxy-configuration-ssl
 
+- name: create configuration folder
+  file:
+    path: "{{ haproxy_cfg_path }}"
+    state: directory
+    owner: root
+    group: root
+    mode: 0750
+  tags:
+    - configuration
+    - haproxy
+    - haproxy-configuration
+
 - name: update configuration file
   template:
     src: etc/haproxy/haproxy.cfg.j2
-    dest: /etc/haproxy/haproxy.cfg
+    dest: "{{ haproxy_main_cfg_part }}"
     owner: root
     group: root
     mode: 0640
-    validate: 'haproxy -f %s -c'
+    validate: "{{ haproxy_validation_command }}"
+  notify: restart haproxy
```

Comment:
This notification is unnecessary. Only the `assemble` tasks below need to notify the reload handler.